packages/labs/src/components/select/select.tsx

```diff
@@ -264,7 +271,14 @@ export class Select<T> extends React.Component<ISelectProps<T>, ISelectState<T>>
     };
 
     private handleItemSelect = (item: T, event: React.SyntheticEvent<HTMLElement>) => {
-        this.setState({ isOpen: false });
+        const { closeOnSelect = true } = this.props;
+
+        if (!closeOnSelect) {
```

Comment:
@ethanbond for my SA, why is this not enough?
```ts
if (closeOnSelect) {
  this.setState({ isOpen: false });
}
```

Reply:
yeah I'm also curious about this.

also please reverse the conditions so that `if (closeOnSelect) {` reads first.

Reply:
I too agree, this is suspect. why cancel event propagation in the case where the component does not respond?

Reply:
The reason is because even if the `closeOnSelect` condition doesn't get called, the click will propagate through to the popover (or maybe the thing beneath the popover?) which will then cause the popover to close.

There might be a more Blueprint-y way to manage that, happy to try whatever if you have pointers!

Reply:
there's a prop on `MenuItem` that we should try to leverage I think?
 https://github.com/ethanbond/blueprint/blob/bfbf4657d01950a1f1b7300b677aa209e3a46de5/packages/core/src/components/menu/menuItem.tsx#L37

Reply:
Ah! Well in that case... I'll just use that for the specific thing I'm trying to do rather than add a prop to the `<Select />` at all and close this. Good with everyone?